dev/tab1tha/k_nn.py

```diff
@@ -0,0 +1,53 @@
+from sklearn.neighbors import KNeighborsClassifier
+import matplotlib.pyplot as plt
+import numpy as np
+
+def k_nearest(data_train, target_train, data_test, target_test):
+    #training model using the k nearest neighbor algorithm
+    #note that 'neighbor' must be spelt without including a 'u' in order for it to work
+    '''This function trains the model then uses it to predict target values of the test data. In it, the accuracy
+    of the K-nearest neighbor algorithm on this data set is computed and returned '''
+    knn = KNeighborsClassifier(n_neighbors = 5)
+    y = target_train 
+    X = data_train
+
+    #Use data to train the model
+    knn.fit(X,y)
+
+    #predict for the test set
+    target_pred = knn.predict(data_test.values)
+    #compute accuracy and return it
+    return knn.score(data_test, target_test)
+
+def visual_compare(data_train, target_train, data_test, target_test):
+    '''This function outputs a graph of the predictions gotten from the K-nn algorithm. In that same graph, the 
+    actual values are plotted. The variation between the predicted values and actual values is clearly visualised such
+    that the user can choose the optimal value of the n_neighbors parameter of KNeighborsClassifier that will yied maximum
+    prediction accuracy'''
+    # Setup arrays to store train and test accuracies
+    neighbors = np.arange(1, 9)
```

Comment:
nit: Did you experiment with values beyond the 1:9 range?
Please specify the motivation of this range in particular.

Reply:
Increasing the range only increased the scope, such that if there is another appropriate point, later in the number line, the user will be able to see that too. These values were however chosen randomly, respecting the requirements that values of n_neighbors must be greater than zero and be of integer type. More is done in the new commit, with the help of hyperparameter tuning.